lib/routes/api/applications.js

```diff
@@ -104,7 +104,7 @@ router.get('/:sid', async(req, res) => {
     const account_sid = req.user.hasAccountAuth ? req.user.account_sid : null;
     const results = await Application.retrieve(req.params.sid, service_provider_sid, account_sid);
     if (results.length === 0) return res.status(404).end();
-    return res.status(200).json(results);
+    return res.status(200).json(results[0]);
```

Comment:
In all fairness I feel like the SQL query should be returning a single entry so this feels like maybe also a hack?

Reply:
Funny you mention that, all of `retrieve()` methods every routes happen to do this. Only one, until now, doesn't have and that is `Application`

Reply:
Ya I believe I'm thinking about this wrong lol, at least at the SQL level. So actually this makes sense lol. I suppose the queries will always technically return rows and its up to the application code to return all of them or just one of them like you have here. Sorry, my brain was not quite ready for SQL at the moment lol.

Reply:
Haha. I have no idea how SQL works.

Reply:
All I know is select table and see what is going on.

Reply:
Feel free to push the changes on the webapp refresh branch and I'll take a look at things. I have this api server branch checked out locally so I can test properly 👍